### DISCUSSION TITLE
How do I redirect www to non-www

### DISCUSSION THREAD

**Discussion by younes-alouani** [vercel/next.js#42523] [Help] [answered]
What is the best way to redirect www to non-www ?

I use Next JS as a container on AWS (ECS + ROUTE 53)

**Comment by icyJoseph** [comment, marked as answer]
I think this is sort of an overlapping concern, if you have some kind of server handling incoming requests that one could change, `www` to `non-www`,  but I guess somehow Next.js could also do this. I found this answer: https://github.com/vercel/next.js/discussions/33554#discussioncomment-2030835, basically match the incoming request in middleware and if it has `www` redirect to `non-www`.

Since middleware runs next to the Next.js server, in non-edge supporting fields, like a raw ECS for example, then you are good to go. That is to say that middleware will just like it does on your local machine.

**Reply by younes-alouani** [reply]
Thank you for the explanations.
I'm thinking about outsourcing this to my load balancer since I have one in front of my ec2 instances auto scaling group. Thus i will add a rule to control the redirections :
- http to https
- www to non www

**Reply by icyJoseph** [reply]
Ah exactly a load balancer sounds more appropriate, keeping concerns separated! 

Don't forget to mark as answered, [#33554 (comment)](https://github.com/vercel/next.js/discussions/33554#discussioncomment-2030835) was also interesting to see.
